controller/lib/hal/hal_stm32.cpp

```diff
@@ -341,12 +342,31 @@ void HalApi::startLoopTimer(const Duration &period, void (*callback)(void *),
   EnableInterrupt(InterruptVector::TIMER15, IntPriority::LOW);
 }
 
+static float latency, max_latency, loop_time;
+static DebugVar d1("loop_latency", &latency, "Latency of loop function, usec",
+                   "%.2f");
+static DebugVar d2("max_latency", &max_latency,
+                   "Maximum latency of loop function, usec", "%.2f");
+static DebugVar d3("loop_time", &loop_time, "Duration of loop function, usec",
+                   "%.2f");
+
 static void Timer15ISR() {
+  uint32_t start = TIMER15_BASE->counter;
   TIMER15_BASE->status = 0;
 
+  // Keep track of loop latency in uSec
+  // Also max latency since it was last zeroed
+  latency = static_cast<float>(start) * (1.0f / CPU_FREQ_MHZ);
+  max_latency = 0;
```

Comment:
Is this line a bug?

Reply:
Oops, yes!  I know exactly how I did that, thanks for catching it.